cmd/cli/main.go

```diff
@@ -0,0 +1,593 @@
+		return err
+	}
+
+	options, err := processCommand(td, ep, netConfigs)
+	if err != nil {
+		return err
+	}
+
+	if argsConfig.withFunding {
+		err = fundWallets(td, ep, netConfigs)
+		if err != nil {
+			return err
+		}
+	}
+
+	return generateAndSendTransaction(options, ep)
+}
+
+func processCommand(td *testData, proxy workflows.ProxyHandler, config *data.NetworkConfig) (*selectedOptions, error) {
+	options, err := getDefaultOptions(td, proxy, config)
+	if err != nil {
+		return nil, err
+	}
+
+	err = setSenderOption(td, options)
+	if err != nil {
+		return nil, err
+	}
+
+	err = setReceiverOption(td, options)
+	if err != nil {
+		return nil, err
+	}
+
+	err = setGuardianOption(td, options)
+	if err != nil {
+		return nil, err
+	}
+
+	err = setGuardedTxByOption(td, options)
+	if err != nil {
+		return nil, err
+	}
+
+	err = setOtherOptions(options, config)
+	if err != nil {
+		return nil, err
+	}
+
+	return options, nil
+}
+
+func setSenderOption(td *testData, options *selectedOptions) error {
+	selectedAddress, sk, err := selectAddressAndSkFromString(td, argsConfig.sender)
+	if err != nil {
+		return err
+	}
+
+	if selectedAddress != nil {
+		options.txArguments.SndAddr = selectedAddress.AddressAsBech32String()
+		options.skSender = sk
+	}
+
+	return nil
+}
+
+func setReceiverOption(td *testData, options *selectedOptions) error {
+	selectedAddress, _, err := selectAddressAndSkFromString(td, argsConfig.receiver)
+	if err != nil {
+		return err
+	}
+	if selectedAddress != nil {
+		options.txArguments.RcvAddr = selectedAddress.AddressAsBech32String()
+	}
+
+	return nil
+}
+
+func setGuardianOption(td *testData, options *selectedOptions) error {
+	var err error
+	selectedAddress, _, err := selectAddressAndSkFromString(td, argsConfig.guardian)
+	if err != nil {
+		return err
+	}
+
+	if selectedAddress != nil {
+		options.guardianAddress = selectedAddress
+	}
+	return err
+}
+
+func setGuardedTxByOption(td *testData, options *selectedOptions) error {
+	selectedAddress, sk, err := selectAddressAndSkFromString(td, argsConfig.guardedTxBy)
+	if err != nil {
+		return err
+	}
+
+	if selectedAddress != nil {
+		options.txArguments.GuardianAddr = selectedAddress.AddressAsBech32String()
+		options.skGuardian = sk
+		options.txArguments.Options = maskGuardedTx
+	}
+
+	return nil
+}
+
+func selectAddressAndSkFromString(td *testData, option string) (core.AddressHandler, []byte, error) {
+	var selectedAddress core.AddressHandler
+	var sk []byte
+
+	switch option {
+	case alice:
+		selectedAddress = td.addressAlice
+		sk = td.skAlice
+	case bob:
+		selectedAddress = td.addressBob
+		sk = td.skBob
+	case charlie:
+		selectedAddress = td.addressCharlie
+		sk = td.skCharlie
+	case eve:
+		selectedAddress = td.addressEve
+		sk = td.skEve
+	default:
+		var err error
+		if len(option) > 0 {
+			selectedAddress, err = data.NewAddressFromBech32String(option)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+	}
+
+	return selectedAddress, sk, nil
```

Comment:
for a provided, custom, address, you return the sk as nil here. (default case branch) is this ok?

Reply:
In this case yes, if you use custom addresses, we do not have the private key for that so we return nil.
For the supported addresses (alice, bob, charlie, eve, etc) we have the private keys as well.